kj-rs/own.rs

```diff
@@ -1,18 +1,51 @@
 //! The `workerd-cxx` module containing the [`Own<T>`] type, which is bindings to the `kj::Own<T>` C++ type
 
 use static_assertions::{assert_eq_align, assert_eq_size};
+use std::{fmt, marker::PhantomData};
 
 assert_eq_size!(repr::Own<()>, [*const (); 2]);
 assert_eq_align!(repr::Own<()>, *const ());
 
+/// When we want to use an `Own`, we want the guarantee of being not null only
+/// in direct `Own<T>`, not Maybe<Own<T>>, and using a [`NonNull`] in `Own`
+/// but allowing Nulls for Niche Value Optimization is undefined behavior.
```

Comment:
This makes me curious, do you know how `Option<Box<T>>` handles this? I would guess it encounters the same issues.